src/actions/index.js

```diff
@@ -560,180 +561,189 @@ function updateOutput(
+                                                newProps[componentIdAndProp] = {
+                                                    id: child.props.id,
+                                                    props: {
+                                                        [childProp]:
+                                                            child.props[
+                                                                childProp
+                                                            ],
+                                                    },
+                                                };
+                                            }
+                                        });
+                                    }
+                                }
+                            );
 
-                    /*
-                     * In some cases, the new item will be an output
-                     * with its inputs already rendered (not rendered)
-                     * as part of this update.
-                     * For example, a tab with global controls that
-                     * renders different content containers without any
-                     * additional inputs.
-                     *
-                     * In that case, we'll call `updateOutput` with that output
-                     * and just "pretend" that one if its inputs changed.
-                     *
-                     * If we ever add logic that informs the user on
-                     * "which input changed", we'll have to account for this
-                     * special case (no input changed?)
-                     */
+                            /*
+                         * Organize props by shared outputs so that we
+                         * only make one request per output component
+                         * (even if there are multiple inputs).
+                         *
+                         * For example, we might render 10 inputs that control
+                         * a single output. If that is the case, we only want
+                         * to make a single call, not 10 calls.
+                         */
 
-                    const outputIds = [];
-                    keys(newProps).forEach(idAndProp => {
-                        if (
-                            // It's an output
-                            InputGraph.dependenciesOf(idAndProp).length === 0 &&
                             /*
-                             * And none of its inputs are generated in this
-                             * request
-                             */
-                            intersection(
-                                InputGraph.dependantsOf(idAndProp),
-                                keys(newProps)
-                            ).length === 0
-                        ) {
-                            outputIds.push(idAndProp);
-                            delete newProps[idAndProp];
```

Comment:
It's pretty hard for me to read the changes in this file - A lot of code hasn't changed, it was just moved and indented. When this is ready for review, could we break up these changes into several commits, so that I can more easily determine which lines were changed vs just moved?

Reply:
That is prettier fault, it completely messes up when there's wrong indent mixed up.

Reply:
I can't tell if you are being sarcastic or not but in case you are not... prettier doesn't run git or push code to Github and it isn't responsible for making PRs pleasant to review. You are responsible for these things.

My usual workflow when developing on un-pretty things is to open the unpretty file, add a space, save... so that prettier runs. Then I commit that as "prettify index.js". Then subsequent commits make your actual logical commits. 

That way a reviewer can step through your PR commit by commit like a book! It's the developers responsibility to make sure reviewing is good experience :)

Also git rebase and rewrite a messy history so your PR has a nice sequence of commits that can be stepped through. Add longer commit descriptions where appropriate. 

Spend time making the life of your reviewers pleasant!

Reply:
Sorry, should have committed before running prettier. Thought it would only add missing `,`, but the comments had bad indent so it messed up.